src/Symfony/Component/Yaml/Command/LintCommand.php

```diff
@@ -137,40 +143,75 @@ private function display(SymfonyStyle $io, array $files): int
                 return $this->displayTxt($io, $files);
             case 'json':
                 return $this->displayJson($io, $files);
+            case 'github':
+                return $this->displayTxt($io, $files, true);
             default:
                 throw new InvalidArgumentException(sprintf('The format "%s" is not supported.', $this->format));
         }
     }
 
-    private function displayTxt(SymfonyStyle $io, array $filesInfo): int
+    private function displayTxt(SymfonyStyle $io, array $filesInfo, bool $errorAsGithubAnnotations = false): int
     {
         $countFiles = \count($filesInfo);
         $erroredFiles = 0;
         $suggestTagOption = false;
 
+        $buffer = [];
         foreach ($filesInfo as $info) {
             if ($info['valid'] && $this->displayCorrectFiles) {
                 $io->comment('<info>OK</info>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
             } elseif (!$info['valid']) {
                 ++$erroredFiles;
-                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
-                $io->text(sprintf('<error> >> %s</error>', $info['message']));
 
                 if (false !== strpos($info['message'], 'PARSE_CUSTOM_TAGS')) {
                     $suggestTagOption = true;
                 }
+
+                if ($errorAsGithubAnnotations) {
+                    $buffer[] = $this->formatErrorAsGithubAnnotation($info);
+                    continue;
+                }
+
+                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
+                $io->text(sprintf('<error> >> %s</error>', $info['message']));
             }
         }
 
         if (0 === $erroredFiles) {
             $io->success(sprintf('All %d YAML files contain valid syntax.', $countFiles));
         } else {
-            $io->warning(sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : ''));
+            $message = sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : '');
+
+            if ($errorAsGithubAnnotations) {
+                $io->writeln(sprintf('::warning::%s', $message));
```

Comment:
Do we even need the summary as a warning here ? If they want to see full details, the logs are there in the checks tab. And errors are annotation (and the Github UI already displays the count of annotations IIRC)

Reply:
No strong opinion.
Github displays the count, but it's per workflow, not per step.

Reply:
and I would say that you won't care about the exact number of invalid YAML files most of the time (you care about the fact there are some, and errors are here for that already). If you care about the exact number, you can still go check the raw logs in the checks tab, which contains that info.

Reply:
note also that putting this warning here consumes one annotation from the limit of annotations (and it won't be the one being dropped, as it is the first one).

Reply:
> and it won't be the one being dropped, as it is the first one

Which was intended. Also IIUC, the warning and error annotations limits are distinct (10 each).

But alright, let's drop it 👌